mantidproject/mantid
# Workbench: make closing the save and overwrite dialog behave correctly

**Description of work.**
#26516 made it so that when the dialog window asking if you want to overwrite the existing file appeared, clicking the X in the corner would wrongly cause the dialog for choosing file location to appear. This has been fixed by adding a cancel button to the dialog. Some code which appeared to never be executed has also been removed.

**To test:**
1. Open an existing project in Workbench.
2. Make a change, e.g. load a new workspace.
3. Click the X in the top-right corner to close Workbench.
4. When asked if you want to save changes, click yes.
5. When asked if you want to overwrite, check that there is a Cancel button, and that clicking the Cancel button or the X in the corner of the dialog cause the window to close and nothing else happens.

Fixes #26513

No release notes because the issue didn't exist previously.

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
